### PR TITLE
stx: explainTransaction and signTransaction

### PR DESCRIPTION
**Pull request by fariedt**
Two methods for the `modules/core` implementation.

### CODE REVIEW

File: modules/core/src/v2/coins/stx.ts [hunk 1]
```diff
@@ -15,6 +24,35 @@ interface SupplementGenerateWalletOptions {
...
+  contractAddress?: string;
+  contractName?: string;
+  contractFunction?: string;
+  contractFunctionArgs?: { type: string; value: string }[];
+}
+
+export interface ExplainTransactionOptions {
+  txHex?: string; // txHex is poorly named here; it is just a wrapped JSON object
+  halfSigned?: {
+    txHex: string; // txHex is poorly named here; it is just a wrapped JSON object
```
Comment (wesgraham) [inline modules/core/src/v2/coins/stx.ts:42]: rename: txJson?

File: modules/core/src/v2/coins/stx.ts [hunk 2]
```diff
@@ -110,10 +148,121 @@ export class Stx extends BaseCoin {
...
+   * @param params
+   * @param callback
+   */
+  signTransaction(
+    params: StxSignTransactionOptions,
+    callback?: NodeCallback<SignedTransaction>
+  ): Bluebird<SignedTransaction> {
+    const self = this;
+
+    return co<SignedTransaction>(function*() {
```
Comment (wesgraham) [inline modules/core/src/v2/coins/stx.ts:162]: can we not use async/await here?

File: modules/core/src/v2/coins/stx.ts [hunk 3]
```diff
@@ -110,10 +148,121 @@ export class Stx extends BaseCoin {
...
+   * Explain a Stacks transaction from txHex
+   * @param params
+   * @param callback
+   */
+  explainTransaction(
+    params: ExplainTransactionOptions,
+    callback?: NodeCallback<StxTransactionExplanation>
+  ): Bluebird<StxTransactionExplanation> {
+    const self = this;
+    return co<TransactionExplanation>(function*() {
```
Comment (wesgraham) [inline modules/core/src/v2/coins/stx.ts:196]: same async/await

File: modules/core/src/v2/coins/stx.ts [hunk 4]
```diff
@@ -110,10 +148,121 @@ export class Stx extends BaseCoin {
...
+      if (tx.type === accountLib.BaseCoin.TransactionType.Send) {
+        const outputs: TransactionRecipient[] = [
+          {
+            address: txJson.payload.to,
+            amount: txJson.payload.amount,
+            memo: txJson.payload.memo,
+          },
+        ];
+
+        const displayOrder = ['id', 'outputAmount', 'changeAmount', 'outputs', 'changeOutputs', 'fee', 'type'];
```
Comment (wesgraham) [inline modules/core/src/v2/coins/stx.ts:221]: can you add memo

File: modules/core/src/v2/coins/stx.ts [hunk 5]
```diff
@@ -110,10 +148,121 @@ export class Stx extends BaseCoin {
...
+
+        return explanationResult;
+      }
+
+      if (tx.type === accountLib.BaseCoin.TransactionType.ContractCall) {
+        const displayOrder = [
+          'id',
+          'fee',
+          'type',
+          'contractAddress',
```
Comment (wesgraham) [inline modules/core/src/v2/coins/stx.ts:242]: can you add memo as well

File: modules/core/src/v2/coins/stx.ts (hunk 2 above)
Comment (tylerlevine) [reply modules/core/src/v2/coins/stx.ts:162]: No async await in implementation code yet please - we need to return a consistent promise type.

File: modules/core/src/v2/coins/stx.ts (hunk 1 above)
Comment (fariedt) [reply modules/core/src/v2/coins/stx.ts:42]: The comments are from the code copied from the `trx.ts` coin.  It's not accurate -- the `txHex` field really is a hex string, not JSON.  I'll remove the comments.

**Review by wesgraham — APPROVED** [review]

**Pull request merged**